src/settings.cpp

```diff
@@ -32,10 +33,110 @@ settings::settings()
     no_retarget_proof_of_work_limit(0x207fffff),
     min_timespan(target_timespan_seconds / retargeting_factor),
     max_timespan(target_timespan_seconds * retargeting_factor),
-    retargeting_interval(target_timespan_seconds / target_spacing_seconds)
+    retargeting_interval(target_timespan_seconds / target_spacing_seconds),
+    genesis_block(*this)
 {
 }
 
+settings::settings(config::settings context)
```

Comment:
The use of text deserialization internal to the settings instance is inconsistent. Setting and command line deserialization is implemented in `bc::config` classes, often encapsulating the primitive. Additionally, use of genesis(text) below is unnecessary. There is no reason to construct the object from a text string. The text stream is broken up into (sort of) logical values for readability. But this is much more clearly and safely accomplished using the `chain::block` constructor as it is intended. Comparing the result to a well-known text stream and block hash result is the job of two text cases.

Reply:
Will fix.

Reply:
I have removed the `genesis()` helper method. I will add the deserialization class in `bc::config` in an upcoming PR to prevent this one from growing too much.